document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/service/impl/AccountsServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.document.generator.accounts.service.impl;
+
+import static uk.gov.companieshouse.document.generator.accounts.AccountsDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.accounts.Accounts;
+import uk.gov.companieshouse.document.generator.accounts.data.accounts.AccountsManager;
+import uk.gov.companieshouse.document.generator.accounts.exception.ServiceException;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Service
+public class AccountsServiceImpl implements AccountsService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    @Autowired
+    private AccountsManager accountsManager;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Accounts getAccounts(String resource) throws ServiceException {
+        LOG.info("Getting accounts data: " + resource);
+
+        Accounts accounts;
+        try {
+            accounts = accountsManager.getAccounts(resource);
+        } catch (Exception e) {
+            LOG.error(e);
+            throw new ServiceException(e.getMessage(), e.getCause());
+        }
+
+        LOG.trace("Accounts data retrieved successfully: " + resource);
+        return accounts;
```

Comment:
The return could be performed inside the `try` clause to save assigning a temporary variable `accounts`

Reply:
It was more for the logging on line 37, but I'll remove it for now and do the one line return as the logging is still to be refined.